vscode/src/azure/workspaceActions.ts

```diff
@@ -115,114 +70,135 @@ workspace = azure.quantum.Workspace(
-  // For the MSA case, you need to query the tenants first and get the underlying AzureAD
-  // tenant for the 'guest' MSA. See https://stackoverflow.microsoft.com/a/76246/108570
-  const firstAuth = await getAuthSession([scopes.armMgmt], associationId);
+  const connStr = await vscode.window.showInputBox({
+    prompt: "Enter the connection string",
+    placeHolder:
+      "SubscriptionId=<guid>;ResourceGroupName=<name>;WorkspaceName=<name>;ApiKey=<secret>;QuantumEndpoint=<serviceUri>;",
+  });
+  if (!connStr) return;
 
-  if (!firstAuth) {
-    log.error("No authentication session returned");
-    sendTelemetryEvent(
-      EventType.QueryWorkspacesEnd,
-      {
-        associationId,
-        reason: "no auth session returned",
-        flowStatus: UserFlowStatus.Failed,
-      },
-      { timeToCompleteMs: performance.now() - start },
+  const parts = connStr.split(";");
+  if (!parts) {
+    vscode.window.showErrorMessage(
+      "Invalid connection string. Please follow the placeholder format",
     );
     return;
   }
 
-  const firstToken = firstAuth.accessToken;
-  const azureUris = new AzureUris();
+  const partsMap = new Map<string, string>();
+  parts.forEach((part) => {
+    const eq = part.indexOf("=");
+    if (eq === -1) return;
+    partsMap.set(part.substring(0, eq).toLowerCase(), part.substring(eq + 1));
+  });
+
+  if (
+    !partsMap.has("subscriptionid") ||
+    !partsMap.has("resourcegroupname") ||
+    !partsMap.has("workspacename") ||
+    !partsMap.has("apikey") ||
+    !partsMap.has("quantumendpoint")
+  ) {
+    vscode.window.showErrorMessage(
+      "Invalid connection string. Please follow the placeholder format",
+    );
+    return;
+  }
 
-  const tenants: ResponseTypes.TenantList = await azureRequest(
-    azureUris.tenants(),
-    firstToken,
+  const workspaceId =
+    `/subscriptions/${partsMap.get("subscriptionid")}` +
+    `/resourceGroups/${partsMap.get("resourcegroupname")}` +
+    `/providers/Microsoft.Quantum/Workspaces/${partsMap.get("workspacename")}`;
+
+  // TODO: Validate the connection string info before returning as valid for further use.
+  // e.g. check for 401 (invalid key), 404 (invalid workspace), endpoint not found (invalid endpoint), etc.
+
+  return {
+    id: workspaceId,
+    name: partsMap.get("workspacename")!,
+    endpointUri: partsMap.get("quantumendpoint")!,
+    tenantId: "", // Blank means not authenticated via a token
+    apiKey: partsMap.get("apikey"),
+    providers: [], // Providers and jobs will be populated by a following 'queryWorkspace' call
+    jobs: [],
+  };
+}
+
+async function chooseWorkspaceFromSubscription(
+  subId: string,
+  associationId: string,
+  token: string,
+  start: number,
+) {
+  // *** Fetch the Quantum Workspaces in the subscription ***
+  const workspaces: ResponseTypes.WorkspaceList = await azureRequest(
+    AzureUris.workspaces(subId),
+    token,
     associationId,
   );
-  log.trace(`Got tenants: ${JSON.stringify(tenants, null, 2)}`);
```

Comment:
A lot of stuff got moved around in this file, which is really confusing the diff algorithm. Probably best to just look at the new code, not the diff, in these parts.